app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -22,7 +22,12 @@ def insecure
   def endpoint
     return global_registration_url if registration_params['smart_proxy_id'].blank?
 
-    "#{@smart_proxy.url}/register"
+    template_settings = @smart_proxy.smart_proxy_features
+                                    .find_by(feature: Feature.find_by(name: 'Templates'))
+                                    .settings
+    url = template_settings['template_url'].presence || @smart_proxy.url
```

Comment:
There's a much shorter way for this:
```suggestion
    url = @smart_proxy.setting('Templates', 'template_url') || @smart_proxy.url
```